### DISCUSSION TITLE
'Version' object has no attribute 'download'

### DISCUSSION THREAD

**Discussion by andysingal** [roboflow/supervision#718] [Q&A]
### Search before asking

- [X] I have searched the Supervision [issues](https://github.com/roboflow/supervision/issues) and found no similar bug report.

### Bug

```
from roboflow import Roboflow
rf = Roboflow(api_key="QKnIGsOx41j1fVIu70oi")
project = rf.workspace().project("u.s.-coins-dataset-a.tatham")
dataset = project.version(9).download("yolov8")
```
ERROR:
```
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
[<ipython-input-8-4c2cd696353e>](https://localhost:8080/#) in <cell line: 4>()
      2 rf = Roboflow(api_key="QKnIGsOx41j1fVIu70oi")
      3 project = rf.workspace().project("u.s.-coins-dataset-a.tatham")
----> 4 dataset = project.version(8).download("yolov8")

AttributeError: 'Version' object has no attribute 'download'
```

### Environment

google colab
```
!pip install super-gradients roboflow python-dotenv -Uq
```

### Minimal Reproducible Example

```
from roboflow import Roboflow
rf = Roboflow(api_key="QKnIGsOx41j1fVIu70oi")
project = rf.workspace().project("u.s.-coins-dataset-a.tatham")
dataset = project.version(8).download("yolov8")
```
ERROR:
```
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
[<ipython-input-8-4c2cd696353e>](https://localhost:8080/#) in <cell line: 4>()
      2 rf = Roboflow(api_key="QKnIGsOx41j1fVIu70oi")
      3 project = rf.workspace().project("u.s.-coins-dataset-a.tatham")
----> 4 dataset = project.version(8).download("yolov8")

AttributeError: 'Version' object has no attribute 'download'
```
<img width="1404" alt="Screenshot 2024-01-04 at 10 08 20 AM" src="https://github.com/roboflow/supervision/assets/20493493/6f250810-e03d-4187-8671-b596e21559f8">

### Are you willing to submit a PR?

- [X] Yes I'd like to help by submitting a PR!

**Comment by SkalskiP** [comment]
Hi, @andysingal 👋🏻 ! Technically, that issue is related to `roboflow` pip package, which is supported by a different team, but I'll try to do my best to help.

I located the [project](https://universe.roboflow.com/atathamuscoinsdataset/u.s.-coins-dataset-a.tatham) you are interested in, and I successfully downloaded versions 8 and 9.

I encourage you to use code snippets generated in the app.

![Screenshot 2024-01-04 at 09 37 58](https://github.com/roboflow/supervision/assets/26109316/a44fda67-3086-4c5e-9ed6-d611bf3b1fb1)

Notice it contains `workspace_id`. 

![Screenshot 2024-01-04 at 09 40 21](https://github.com/roboflow/supervision/assets/26109316/388abb12-905e-4062-bdd4-f6604578d27a)

Could you follow those steps and check if the issue persists?

**Comment by SkalskiP** [comment]
Hi, @andysingal 👋🏻 ! Any updates on this issue?

**Comment by andysingal** [comment]
> from roboflow import Roboflow
> rf = Roboflow(api_key="QKnIGsOx41j1fVIu70oi")
> project = rf.workspace().project("u.s.-coins-dataset-a.tatham")
> dataset = project.version(8).download("yolov8")

Hi @SkalskiP , sorry for the late reply. 
I tried on Colab Notebook using Runtime: T4: 
<img width="617" alt="Screenshot 2024-01-08 at 10 32 38 AM" src="https://github.com/roboflow/supervision/assets/20493493/dde7d895-d3cf-41be-9fcc-415ccfc752b3">

<img width="1124" alt="Screenshot 2024-01-08 at 10 33 20 AM" src="https://github.com/roboflow/supervision/assets/20493493/4b012135-1ad9-41b2-9b62-ebc4d146c460">

Do not understand what i am missing

**Comment by SkalskiP** [comment]
@andysingal, well now I'm getting confused. 😃 I'll pass that issue along to the engineers who maintain `roboflow` pip package.

**Comment by tonylampada** [comment]
@andysingal 

from your last screenshot it looks like you're trying to download a project that doesn't exist (workspace=first-yolov8, project=us-coins-mlo6t.tatham) [https://universe.roboflow.com/first-yolov8/us-coins-mlo6t.tatham](https://universe.roboflow.com/first-yolov8/us-coins-mlo6t.tatham)

There's a different project with a similar projectID tough: [https://universe.roboflow.com/atathamuscoinsdataset/u.s.-coins-dataset-a.tatham](https://universe.roboflow.com/atathamuscoinsdataset/u.s.-coins-dataset-a.tatham)

Is this the one you're trying to download?

If so, you might wanna try the suggested snippet on its Roboflow Universe page:

![CleanShot 2024-01-08 at 12 50 04@2x](https://github.com/roboflow/supervision/assets/218821/e9782d17-b6e8-4222-b2e5-af2a3e9f247e)

**Comment by tonylampada** [comment]
Upon further investigation, the error message you see seems weird and not like what you should be seeing.
We found out that a version pinning in roboflow was making pip download a very old roboflow version when installed together with super-gradients.
This is now fixed on roboflow==1.1.16 so if you run your code again you should see a better error message like

![image](https://github.com/roboflow/supervision/assets/218821/0e02dbf6-0c33-410f-946f-78d9361dd30d)

which would probably have given you a better hint to find the problem in the first place :-)

Cheers.

**Comment by onuralpszr** [comment]
Converting this issue to Q&A
